controllers/flowlogspipeline/flp_objects.go

```diff
@@ -252,13 +265,83 @@ func (b *builder) addTransformStages(lastStage *config.PipelineBuilderStage) {
 	}
 
 	// prometheus stage (encode) configuration
+	// the aggregate and prometheus rules will be incorporated from the metrics definitions file
 	agg := enrichedStage.Aggregate("aggregate", []api.AggregateDefinition{})
 	agg.EncodePrometheus("prometheus", api.PromEncode{
 		Port:   int(b.desired.PrometheusPort),
-		Prefix: "flp_",
+		Prefix: "netobserv_",
 	})
 }
 
+var generateStages = []string{"extract_aggregate", "encode_prom"}
+
+func (b *builder) mergeMetricsConfiguration(pipeline config.PipelineBuilderStage) {
+	// set confgen.Opt.SrcFolder, etc
+	confgen.Opt.SrcFolder = b.desired.MetricsDefinitionsDir
+	confgen.Opt.DestConfFile = "/dev/null"
+	confgen.Opt.SkipWithTags = b.desired.IgnoreMetrics
+	confgen.Opt.GenerateStages = generateStages
+
+	if confgen.Opt.SrcFolder == "" {
+		return
+	}
+	// run the confgenerator to produce the proper flp configuration for metrics from metrics_definitions
+	cg, err := confgen.NewConfGen()
+	if err != nil {
+		log.Fatalf("failed to initialize NewConfGen %s", err)
+		return
+	}
+	err = cg.Run()
+	if err != nil {
+		log.Fatalf("failed to initialize NewConfGen %s", err)
+		return
+	}
+	truncatedConfig := cg.GenerateTruncatedConfig(generateStages)
+	yFileBytes, err := yaml.Marshal(truncatedConfig)
+	if err != nil {
+		log.Fatal(err)
+		return
+	}
+	var metricsConfig struct {
+		Parameters []config.StageParam
+	}
+	err2 := yaml.Unmarshal(yFileBytes, &metricsConfig)
+	if err2 != nil {
+		log.Fatal(err2)
+		return
+	}
+
+	// merge the new items into the existing parameters configuration
+	parameters := pipeline.GetStageParams()
+
+	// obtain pointers to various parameters structures:
+	var ea2 []api.AggregateDefinition
+	var pe1, pe2 *api.PromEncode
+	var extractIndex int
+
+	// we build upon the fact that in addTransformStages there are defined stages
+	// Transform.Network, Extract.Aggregates, and Encode.Prom
+	// Find the parameter sections for the Extract.Aggregates, and Encode.Prom stages
+	for i, p := range parameters {
+		if p.Extract != nil && p.Extract.Aggregates != nil {
+			extractIndex = i
+		}
+		if p.Encode != nil && p.Encode.Prom != nil {
+			pe1 = p.Encode.Prom
+		}
+	}
+	for _, p := range metricsConfig.Parameters {
+		if p.Extract != nil && p.Extract.Aggregates != nil {
+			ea2 = p.Extract.Aggregates
+		}
+		if p.Encode != nil && p.Encode.Prom != nil {
+			pe2 = p.Encode.Prom
+		}
+	}
+	parameters[extractIndex].Extract.Aggregates = ea2
+	pe1.Metrics = pe2.Metrics
```

Comment:
We could simplify by removing this merging logic if the confgen was performed before creating the pipeline's Aggregate and Prom stages, directly called from `addTransformStages` function. 

Could be something like that (in `addTransformStages`):

```go
	confgenAgg, confgenProm := b.getMetricsConfiguration()
	agg := enrichedStage.Aggregate("aggregate", confgenAgg)
	agg.EncodePrometheus("prometheus", api.PromEncode{
		Port:    int(b.desired.PrometheusPort),
		Prefix:  "netobserv_",
		Metrics: confgenProm.Metrics,
	})
```

Reply:
done